src/lib/provable/gadgets/sha256.ts

```diff
@@ -287,3 +229,87 @@ function sigma(u: UInt32, bits: TupleN<number, 3>, firstShifted = false) {
     .xor(UInt32.Unsafe.fromField(xRotR1))
     .xor(UInt32.Unsafe.fromField(xRotR2));
 }
+
+/**
+ * Performs the SHA-256 compression function on the given hash values and message schedule.
+ *
+ * @param H - The initial or intermediate hash values (8-element array of UInt32).
+ * @param W - The message schedule (64-element array of UInt32).
+ * @param K - The constants used in the SHA-256 algorithm (64-element array of UInt32).
+ * @returns The updated intermediate hash values after compression.
+ */
+function sha256Compression(H: UInt32[], W: UInt32[], K: UInt32[]) {
```

Comment:
note: this is the original code just moved

Reply:
Yes it is, should I add a notice in JSDoc?
It's also the case for `computeMessageSchedule`

Reply:
nope all good!